### PR TITLE
Added `copy_from` service to `NativeString`

### PR DESCRIPTION
**Pull request by R4PaSs**
Usability PR, since I got tired of handling indexes and different structures manually, this method abstracts the job of copying the content of a `Text` to a `NativeString` object.

### CODE REVIEW

File: lib/core/text/abstract_text.nit [hunk 1]
```diff
@@ -1095,6 +1095,13 @@ abstract class Text
 		end
 	end
 
+	# Copies the content of `self` to `dst`
+	#
+	# Easier, more user-friendly version of `copy_to_native`
+	#
+	# NOTE: `dst` must be large enough to withold `self.bytelen` bytes
+	fun copy_to(dst: NativeString) do copy_to_native(dst, bytelen, 0, 0)
```
Comment (privat) [inline lib/core/text/abstract_text.nit]: I have no problem with the idea but I do not like the name. I think it is quite dangerous (not POLA etc) to expose a low-level service (everything with NativeString is low-level) in a high-level class (Text) with a very generic name (copy_to).

Comment (R4PaSs) [inline lib/core/text/abstract_text.nit]: It was mainly to mimic the copy API from NativeString/NativeArray, I can change it, but I don't have a better idea for a name than this

**Comment by R4PaSs** [comment]
test this please

**Comment by R4PaSs** [comment]
Reroll: Changed service from `Text::copy_to` to `NativeString::copy_from`

**Comment by privat** [comment]
+1

**Comment by xymus** [comment]
Is it like a `String::to_cstring` where you manage you own buffer?

I'm not sure about the name as it sounds really similar to `NativeString::copy_to` while not being symmetric . http://nitlanguage.org/doc/stdlib/property_core__text__native__NativeString__copy_to.html#core__text__native__NativeString__copy_to

I would propose instead `fill_from`, `overwrite_from` or exchanging the receivers `Text::write_to_cstring`.

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
ping. I merge or I wait?

**Comment by R4PaSs** [comment]
I can rename the function to satisfy @xymus, this does not bother me that much

**Comment by R4PaSs** [comment]
Reroll: The method is now named `fill_from`, I took advantage of the reroll to bring more performance to the `Rope` version of `copy_to_native` and to add one more test for this method as well.

**Comment by xymus** [comment]
+1

**Comment by xymus** [comment]
There's a new missing license.

**Comment by R4PaSs** [comment]
Reroll: Fixed missing license

**Pull request merged**